karapace/schema_registry_apis.py

```diff
@@ -52,9 +60,110 @@ class SchemaErrorMessages(Enum):
     )
 
 
+@unique
+class Operation(Enum):
+    Read = "Read"
+    Write = "Write"
+
+
+@dataclass
+class User:
+    username: str
+    password: str = field(repr=False)  # hashed
+
+    def compare_password(self, plaintext: str) -> bool:
+        return compare_digest(self.password, crypt.crypt(plaintext, self.password))
+
+
+@dataclass(frozen=True)
+class ACL:
+    principal: str
+    operation: Operation
+    resource: re
+
+
+def http_authorizer(filename: str):
+    with open(filename, "r") as authfile:
+        try:
+            authdata = json.load(authfile)
+
+            userdb = {user["username"]: User(user["username"], user["password"]) for user in authdata["users"]}
+            acls = [
+                ACL(acl["principal"], Operation(acl["operation"]), re.compile(acl["resource"])) for acl in authdata["acls"]
+            ]
+            log.info(
+                "Loaded schema registry ACL rules: %s",
+                [(acl.principal, acl.operation.value, acl.resource.pattern) for acl in acls],
+            )
+        except Exception as ex:
+            raise InvalidConfiguration("Auth configuration is not a valid JSON") from ex
+
+    def check_authorization(user: User, operation: Operation, resource: str) -> bool:
+        def check_operation(operation: Operation, acl: ACL) -> bool:
+            if operation == Operation.Read:
+                return True
+            if acl.operation == Operation.Write:
+                return True
```

Comment:
Does this mean that write access implies read access?

Reply:
My idea was that write implies read access as it makes no sense to have only write without read...